synapse/storage/persist_events.py

```diff
@@ -223,32 +222,29 @@ def persist_events(
         for room_id in partitioned:
             self._maybe_start_persisting(room_id)
 
-        yield make_deferred_yieldable(
+        await make_deferred_yieldable(
             defer.gatherResults(deferreds, consumeErrors=True)
         )
 
-        max_persisted_id = yield self.main_store.get_current_events_token()
-
-        return max_persisted_id
+        return self.main_store.get_current_events_token()
 
-    @defer.inlineCallbacks
-    def persist_event(
-        self, event: FrozenEvent, context: EventContext, backfilled: bool = False
-    ):
+    async def persist_event(
+        self, event: EventBase, context: EventContext, backfilled: bool = False
+    ) -> Tuple[int, int]:
         """
         Returns:
-            Deferred[Tuple[int, int]]: the stream ordering of ``event``,
-            and the stream ordering of the latest persisted event
+            The stream ordering of ``event``, and the stream ordering of the
+            latest persisted event
         """
         deferred = self._event_persist_queue.add_to_queue(
             event.room_id, [(event, context)], backfilled=backfilled
         )
 
         self._maybe_start_persisting(event.room_id)
 
-        yield make_deferred_yieldable(deferred)
+        await make_deferred_yieldable(deferred)
 
-        max_persisted_id = yield self.main_store.get_current_events_token()
+        max_persisted_id = self.main_store.get_current_events_token()
```

Comment:
So the `yield` here was a mistake, right? `get_current_events_token` or anything that overrides it should never return an `Awaitable`, correct?

Reply:
@erikjohnston and I both believe it was an unnecessary yield: https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$wzYKCOlFuDvdH1HeyVRovkdDEASrRq-6AfSt_QDcVFg?via=matrix.org&via=vector.modular.im&via=privacytools.io

When tracing the functions I couldn't find any that returned awaitables.

Should have mentioned this in the description, sorry!

Reply:
All good, thanks for the explanation!